### DISCUSSION TITLE
Unable to accommodate for 2 devices on the same U in the rack

### DISCUSSION THREAD

**Discussion by svetlinski** [netbox-community/netbox#9053] [Q&A]
### NetBox version

v3.1.6

### Python version

3.8

### Steps to Reproduce

1. Create a patch panel device (1U), **not** full depth
2. Create a switch (1U) full depth
3. Create a rack (42U) default dimensions
4. Add the patch panel device - any U, for example, front facing, U28
5. Try to add the switch device - same U as above (U28), rear facing
6. The switch cannot be added

The patch panel device above has been used for argument's sake. It can also be a cable tidy accessory which is front-facing and has no depth at all when it comes to the inside of the rack. "Patch panel" and "cable tidy" are interchangeable terms in this case. 

The question here is - in real life, the patch panel normally doesn't take much space (depth-wise), especially if there are not many cables connected to it, allowing for another device, in this case - a switch, to be installed in the rear of the rack.
Netbox doesn't allow me to add the switch in the rear of the rack as the switch is full depth. If I change the switch device to be full depth, this will affect all other racks as well, where we want to see the switch devices as full depth. 

I wasn't able to find a way around this and I checked the forum, but I didn't see anything similar, apologies if I have missed something. 

Has this been addressed in Netbox v3.2.0, if not, is there anything I can do or are there plans to introduce a different measure of depth, rather than the existing one? 

Thanks in advance.
![NetboxRackIssue](https://user-images.githubusercontent.com/99681269/161942505-0260b205-2539-4066-b27d-f5a9693d6237.PNG)

### Expected Behavior

To be able to have a patch panel in the rack, front facing and a switch in the rear, at the same U, rear facing.

### Observed Behavior

Netbox is not letting me add the switch in the same U as a patch panel or a cable tidy

**Comment by svetlinski** [comment]
![InkedIMG_0475_LI](https://user-images.githubusercontent.com/99681269/161945700-6561fe70-2ae2-4736-8435-8bd98372929c.jpg)

The above image shows the rear of the rack, where you can see some of the patch panels and cable tidy accessories mounted on the front of the rack and the switches mounted at the rear. 

I can provide further information if needed.

Thanks in advance.

**Comment by phurrelmann** [comment]
If you want 2 devices on the same rack location both devices must not be full-depth.
Full-depth prohibits mount of a second device in the same location and that's fully correct and the wanted behavior.

**Comment by svetlinski** [comment]
Thank you for confirming - I understand the need behind it (no pun intended), and I wanted to know if there's a way to differentiate between a device that is 20cm deep and one that is 25cm. This would then be taken away from the total depth, indicating whether another device can fit in the rack. 
Also, currently, if you put a switch in the rack that is not full depth and check the rack in Netbox, you will be under the impression that you can fit another device in the rear of the rack, while this may not certainly be the case.

**Comment by ghost** [comment]
This is kind of what you're asking for : https://github.com/netbox-community/netbox/issues/1627
It has been brought up before. Personally I think it would be an awesome feature to be able to also track the depths of equipment but likely very difficult to implement.
